### ISSUE TITLE
Seemingly Random Decreased Coverage

### ISSUE THREAD

**Issue by mjb51923** [mjb51923/1_smd13_SNP_calling#7]
This work was all originally done between November 2019 and January 2020, but I have learned more about the best was to align my reads so I am recapitulating parts of what I did before and recording it on GitHub for the first time.

**Comment by mjb51923** [comment]
This work was all originally done between November 2019 and January 2020, but I have learned more about the best was to align my reads so I am recapitulating parts of what I did before and recording it on GitHub for the first time.

**Comment by mjb51923** [comment]
When I originally began looking at these reads I went through several iterations of mapping where I used BWAmem with or without duplicate removal and/or secondary alignment removal. I also used minimap2 to align my short and long reads. After having worked this further I determined that my B73Ab10 (generally referred to as Ab10:30X) and B73:30X should not have duplicates removed while all other short read files should have the duplicates removed. I performed these alignments in the course of my SNP calling and am now going to generate .tdf files with 5000 bp windows for all of these files using the below script titled IGV_Maketdf.sh
```
#PBS -S /bin/bash
#PBS -q batch 
#PBS -N IGV_Maketdf
#PBS -l nodes=1:ppn=1
#PBS -l mem=30gb
#PBS -l walltime=10:00:00

#PBS -M mjb51923@uga.edu
#PBS -m ae

#Load Modules
module load IGVTools/2.3.98-Java-1.8.0_144

#Define Variables
DIR="/scratch/mjb51923/smd13/out/smd13_SNP_calling_v2/"
REF="/scratch/mjb51923/ref_genomes/B73Ab10.pseudomolecules-v2.fasta"

for i in Smd13ThreeReadSets_RG_try2.s.r.bam \
smd8_all_RG.s.r.bam \
smd3_all_RG.s.r.bam \
N10_RG.s.r.bam \
Ab10v2_all_RG.s.r.bam \
Ab10-II_RG.s.r.bam \
Ab10-III_RG.s.r.bam \
smd12_RG.s.r.bam \
smd1_RG.s.r.bam \
J216_round2_RG.s.r.bam \
KS72_round2_RG.s.r.bam \
Delta4-7_RG.s.r.bam \
Ab10:30X_RG.s.bam \
B73:30X_RG.s.bam
do
time igvtools count -w 5000 $DIR/"BWAmem_"$i $DIR/"BWAmem_"$i.tdf $REF
done
```

**Comment by mjb51923** [comment]
It first became apparent that smd13, smd8, and Ab1030X also referred to as B73Ab10 shared a decreased area of coverage at the following two locations on chr10: 186MB-188MB 190MB-191MB. 

To determine if the decreased coverage in these areas of interest was an issue with BWAmem's alignment algorithm I aligned the B73Ab10 reads using Bowtie's default parameters. This produced a similar pattern of coverage so I generated derived reads modeled after Illumina reads. To do this I broke the B73Ab10 reference genome into the same number of 150 bp sections as I had actual reads and aligned them back to the reference genome. This yielded regular coverage giving the impression that the decreased coverage in these areas is not the result of an alignment issue

<img width="960" alt="image" src="https://user-images.githubusercontent.com/58435529/79788993-23682f80-830f-11ea-9482-5b081b702e2f.png">

To determine if the DNA in these regions of decreased coverage look like real maize DNA, Jonathan suggested that I check if the region has normal TE annotations. We did this because at the time the TE annotations were the only one's completed. Jonathan looked at them and said that they appeared typical. This is saved on my local computer as Ab10_TE_annotation.xml

<img width="590" alt="image" src="https://user-images.githubusercontent.com/58435529/79789484-e6e90380-830f-11ea-8894-9645e7d2d32f.png">

**Comment by mjb51923** [comment]
Put stuff about all the ab10 lines here

**Comment by mjb51923** [comment]
Knowing that 190MB-191MB contained some large repeating units we then wanted to know if the decreased coverage was the result of having too many copies that needed to be mapped to on the off chance that for some reason BWAmem was better able to map the derived reads than the real reads. To test this I extracted one single repeat unit from the 5 units to the right of the kindr complex and aligned all of my reads to this small section in the script BWA_map_to_single_unit.sh. I obtained coverage though it wasn't even, this data suggested that repetitive elements may affect the ability of BWAmem to map. This is saved as 1_Right_Rep.xml

<img width="960" alt="image" src="https://user-images.githubusercontent.com/58435529/79790621-a4282b00-8311-11ea-85af-5e45541ffe97.png">

To test if this pattern was consistent I went through the dot plots i had previously made and marked the regions that were conspicuously repetitive, they are the red blocks at the top. I found that 2 of the 3 regions of low coverage on the Ab10 haplotype were in repetitive regions, but one region was not. I also found that 5 repetitive regions on the Ab10 haplotype had normal coverage so it doesn't seem that the decreased coverage is the result of repetitive DNA. This is saved as Reads_From_Jonathan_subset.xml

<img width="855" alt="image" src="https://user-images.githubusercontent.com/58435529/79791277-a9d24080-8312-11ea-8d65-cb336e1b7e48.png">

**Comment by mjb51923** [comment]
To determine where the sequence in the reference genome was coming from I aligned B73Ab10 nanopore and PacBio reads, which were used in the reference genome assembly, back to the reference genome using minimap2 default parameters. The scripts are titled Align_PacBio_Ab10.sh Align_Nanopore_Ab10.sh. The resulting alignments are shown below zoomed in on the area of low coverage directly next to the smd8 deletion

<img width="960" alt="image" src="https://user-images.githubusercontent.com/58435529/79792520-aa6bd680-8314-11ea-9518-86f126b78275.png">

**Comment by mjb51923** [comment]
To determine if these regions of decreased coverage are genome wide I compared the B73Ab10 illumina, nanopore, PacBio, derived short, and derived long reads. I used alignments that had no secondary alignments and a mapping quality greater than or equal to 10. I did this because I wanted to ensure that the derived alignments had low coverage in the difficult areas to map and without filters the derived alignments had mostly uniform coverage. I then did a series of subtractions in order to quantify the change in read counts. ∆NanoporePacBio = read count of Nanopore minus the read count of Pac Bio and ∆NanoporeDLong = read count Nanopore minus read count of derived long. I then took the difference between these so (∆NanoporePacBio-∆NanoporeDLong) to get the difference in the read count between Nanopore and PacBio that cannot be attributed to a difficult region to map to. I did the same thing for the illumina reads so ∆NanoporeIlumina = read count of Nanopore minus the read count of Illumina and ∆NanoporeDShort = read count of Nanopore minus the read cound of the derived short. I then took the difference between those (∆NanoporeIllumina-∆NanoporeDShort) to get the difference in the read count between Nanopore and Illumina that cannot be attributed to a difficult region to map to. I then added (∆NanoporeIllumina-∆NanoporeDShort) + (∆NanoporePacBio-∆NanoporeDLong) in order to pull out areas that had different read coverages in both the Illumina and the PacBio alignemnts. I displayed these as heat maps. and then went through each chromosome and determined regions with high "heat", I did the by eye alone. 

CHR1
<img width="595" alt="image" src="https://user-images.githubusercontent.com/58435529/79913458-3ea76d80-83e9-11ea-8bd4-6723cceb0719.png">

CHR2
<img width="599" alt="image" src="https://user-images.githubusercontent.com/58435529/79913471-4535e500-83e9-11ea-8349-340e03f1dc81.png">

CHR3
<img width="600" alt="image" src="https://user-images.githubusercontent.com/58435529/79913480-49fa9900-83e9-11ea-9c88-2c1300e45b69.png">

CHR4
<img width="541" alt="image" src="https://user-images.githubusercontent.com/58435529/79913502-53840100-83e9-11ea-8521-b21574dbb1e6.png">

CHR5 

<img width="540" alt="image" src="https://user-images.githubusercontent.com/58435529/79913516-5a127880-83e9-11ea-942d-49ec9b69768c.png">

CHR6
<img width="540" alt="image" src="https://user-images.githubusercontent.com/58435529/79913525-5ed72c80-83e9-11ea-8892-d298fd7bf6c0.png">

CHR7
<img width="676" alt="image" src="https://user-images.githubusercontent.com/58435529/79913541-6696d100-83e9-11ea-809e-6f534da72809.png">

CHR8
<img width="673" alt="image" src="https://user-images.githubusercontent.com/58435529/79913556-6dbddf00-83e9-11ea-9c53-0c76844d05b2.png">

CHR9 
<img width="673" alt="image" src="https://user-images.githubusercontent.com/58435529/79913568-731b2980-83e9-11ea-97bd-a57cad9a2355.png">

CHR10
<img width="702" alt="image" src="https://user-images.githubusercontent.com/58435529/79913589-7a423780-83e9-11ea-96bd-3d43d49ed934.png">

I made a table of all of the high "heat" regions, which is below. The regions highlighted in grey have high "heat", but I don't find them to be credible. This leaves me with 18 credible regions of reduced coverage with an average size of 0.689MB and 0.529MB


Chromosome | Start (MB) | End (MB) | Notes
-- | -- | -- | --
2 | 156.5 | 156.8 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
3 | 83.5 | 85 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
3 | 165.9 | 166.3 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
4 | 169.8 | 170 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
4 | 193.1 | 193.15 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
5 | 149 | 152 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
5 | 177.45 | 177.5 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
6 | 105 | 105.5 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
7 | 4.4 | 4.55 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
7 | 93 | 93.5 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
8 | 31.4 | 31.7 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
8 | 31.9 | 32.6 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
8 | 49.5 | 50.2 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
8 | 71.5 | 71.75 | Derived and Nanopore have consistent coverage, PacBio has almost no coverage, illumina has reduced coverage
10 | 2.5 | 3.5 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
10 | 157.1 | 157.4 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
10 | 186.5 | 188 | There are decreases in the derived coverage indicating the region is hard to map to, but there appear to be real difference as well
10 | 190 | 191 | Nanopore and derived have consistent coverage, PacBio and illumina have almost no coverage
5 | 200 | 202.2 | PacBio has higher and more cocsistent coverage than either nanopore or illumina. The derived has some loq quality alignemnts, but generally even coverage
7 | 120 | 120.75 | Nanopore and derived have consistent coverage, PacBio has inconsistent but high coverage, and illumina has low quality coverage
8 | 127.8 | 127.9 | Derives, Nanoporem and PacBio have consistent coverage, Illumina has reduced coverage, I am not confident this real
9 | 76.5 | 77.25 | Derived short and long have decreased coverage in the MQ > 10 filter. I think this is the product of a difficult region to map to

**Comment by mjb51923** [comment]
Next we were wondering if somehow the nanopore read were introducing incorrect sequence. If this were true I would expect chromosomes 1-9 to be longer in the Ab10 genome, which was made with nanopore, than in B73 which was not made with nanopore. I compared the sizes for all of the B73 and B73Ab10 chromosomes and found that for the most part the Ab10 chromsomes are actually smaller than the B73. This suggests nanopore isn't introducing novel DNA


Chromosome | Ab10 (bases) | B73 (Bases) | ∆Ab10_B73 Bases | ∆Ab10_B73 MB
-- | -- | -- | -- | --
chr1 | 308051122 | 308452471 | -401349 | -0.401349
chr2 | 242494172 | 243675191 | -1181019 | -1.181019
chr3 | 236084109 | 238017767 | -1933658 | -1.933658
chr4 | 249748697 | 250330460 | -581763 | -0.581763
chr5 | 225306452 | 226353449 | -1046997 | -1.046997
chr6 | 177079235 | 181357234 | -4277999 | -4.277999
chr7 | 184241495 | 185808916 | -1567421 | -1.567421
chr8 | 182821033 | 182411202 | 409831 | 0.409831
chr9 | 161994764 | 163004744 | -1009980 | -1.00998
chr10 | 195026473 | 152435371 | 42591102 | 42.591102

**Comment by mjb51923** [comment]
Ab10 reference was made using nanopore while B73 was not made using nanopore. We also know that the Ab10 reference genome is in a B73 background so chromosomes 1 through 9 should be the same as B73. So if nanopore is introducing DNA that isn’t present in the pacbio or illumine reads we should see that on chromosomes 1 through 9 B73 reads mapped to the Ab10 reference have regions of low coverage while B73 reads mapped B73 have even coverage. I used the script BWA_align_B73_B73.sh to align B73 reads to the B73 reference genome. I used the B73 read alignment to the B73Ab10 genome that I had generated for small variant calling.

What we see is inconclusive, that sometimes B73 mapped to B73 has the same regions of decreased coverage as B73 mapped to Ab10. 

<img width="1183" alt="Screen Shot 2020-04-22 at 2 21 22 PM" src="https://user-images.githubusercontent.com/58435529/80024401-9bb42980-84a4-11ea-8f1b-ea96c1b551ee.png">
